vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -395,17 +405,56 @@ def build_connector_meta(
                 save_to_host=True,
             )
 
-        meta.reqs_to_send = self._reqs_need_send
-        meta.reqs_in_batch = self._reqs_in_batch
-
         # Clear the list once workers start the transfers
         self._reqs_need_recv.clear()
         self._reqs_need_save.clear()
-        self._reqs_in_batch = set()
-        self._reqs_need_send = {}
 
         return meta
 
+    def update_connector_output(
+        self,
+        connector_output: "KVConnectorOutput",
+    ):
+        finished_sending: set[str] = set()
+
+        # Blocks sent - remove expiry timeout
+        for notif in (connector_output.finished_sending or ()):
+            req_id, tp_ratio = notif.decode("utf-8").rsplit(":", 1)
+            # Sent notifications received after we already timed out
+            if req_id not in self._reqs_need_send:
```

Comment:
Corner case ... we time out (or abort?) and then later receive notifications? Is it possible? We have no test to cover this